lib/WWW/Mechanize.pm

```diff
@@ -834,13 +834,16 @@ sub text {
     my $self = shift;
 
     if ( not defined $self->{text} ) {
-        require HTML::TreeBuilder;
+        unless ( exists $INC{'HTML::TreeBuilder'} ) {
```

Comment:
Why do you skip checking version and importing `-weak` if `exists $INC{'HTML::TreeBuilder'}`?

Reply:
I am checking the version, but it has to be done after the `require` call. It's easy to do it with `use`, but the code we have here explicitly loads the module at run-time. In [the `require` perldoc](http://perldoc.perl.org/functions/use.html) it explains

> If the VERSION argument is present between Module and LIST, then the use will call the VERSION method in class Module with the given version as an argument:
> 
>     use Module 12.34;
> 
> is equivalent to:
> 
>     BEGIN { require Module; Module->VERSION(12.34) }

I put the `%INC` lookup in because without it, it will do the version check and `import` call every single time this method is called. I would have liked to tie it to `require`, but it always returns true even if the module has already been loaded. Under the hood it also checks in `%INC` as far as I understand, so doing that myself before attempting the version check and `import` seemed like the right thing to do.

I'm happy to add a test that proves this works.